Automattic/simplenote-ios
# Charlie/shortcut data loss failsafe

### Fix
With the new append shortcuts there is the possibility that your shortcut fails and if that happens the content you sent to the shortcut will be lost.  There isn't a lot we can do to prevent that unfortunately, but we can attempt to save the content locally and then restore it on the next launch.  So in this PR I have added a content recovery mechanism incase a shortcut fails.  That way the user doesn't have to restore the written content if a shortcut fails.


### Test
1. Build and run simplenote ios
2. Open up the shortcuts app and create an append to note or a create note with content shortcut
3. Disable the internet connections
4. run the shortcut
6. launch simplenote.

 - [ ] Confirm that a new note is created with the content from your shortcut

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
> These changes do not require release notes.
